actors/builtin/market/market_test.go

```diff
@@ -1093,6 +1094,579 @@ func TestOnMinerSectorsTerminate(t *testing.T) {
+				terminationEpoch: abi.ChainEpoch(6),
+				cronTickEpoch:    abi.ChainEpoch(10),
+				payment:          abi.NewTokenAmount(0), // (10 - 10) * 10
+			},
+			"deal is terminated at the activation epoch and then the first crontick happens": {
+				dealStart:        abi.ChainEpoch(10),
+				dealEnd:          abi.ChainEpoch(20),
+				activationEpoch:  abi.ChainEpoch(5),
+				terminationEpoch: abi.ChainEpoch(5),
+				cronTickEpoch:    abi.ChainEpoch(10),
+				payment:          abi.NewTokenAmount(0), // (10 - 10) * 10
+			},
+			"deal is slashed and then deal expiry happens on crontick, but slashing still occurs": {
+				dealStart:        abi.ChainEpoch(10),
+				dealEnd:          abi.ChainEpoch(20),
+				activationEpoch:  abi.ChainEpoch(5),
+				terminationEpoch: abi.ChainEpoch(15),
+				cronTickEpoch:    abi.ChainEpoch(25), // deal has expired
+				payment:          abi.NewTokenAmount(50),
+			},
+			"deal slash epoch must NOT be greater than current epoch": {
+				dealStart:        abi.ChainEpoch(10),
+				dealEnd:          abi.ChainEpoch(20),
+				activationEpoch:  abi.ChainEpoch(5),
+				terminationEpoch: abi.ChainEpoch(15),
+				cronTickEpoch:    abi.ChainEpoch(10), // deal has expired
+				payment:          abi.NewTokenAmount(50),
+				assertionMsg:     "current epoch less than slash epoch",
+			},
+			"deal is slashed just BEFORE the end epoch": {
+				dealStart:        abi.ChainEpoch(10),
+				dealEnd:          abi.ChainEpoch(20),
+				activationEpoch:  abi.ChainEpoch(5),
+				terminationEpoch: abi.ChainEpoch(19),
+				cronTickEpoch:    abi.ChainEpoch(19),
+				payment:          abi.NewTokenAmount(90), // (19 - 10) * 10
+			},
+		}
+
+		for n, tc := range tcs {
+			t.Run(n, func(t *testing.T) {
+				rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+
+				// publish and activate
+				rt.SetEpoch(tc.activationEpoch)
+				dealId := actor.publishAndActivateDeal(rt, client, mAddrs, tc.dealStart, tc.dealEnd, tc.activationEpoch, sectorExpiry)
+				d := actor.getDealProposal(rt, dealId)
+
+				// terminate
+				rt.SetEpoch(tc.terminationEpoch)
+				actor.terminateDeals(rt, provider, dealId)
+
+				//  cron tick
+				rt.SetEpoch(tc.cronTickEpoch)
+
+				if len(tc.assertionMsg) == 0 {
+					pay, slashed := actor.cronTickAndAssertBalances(rt, client, provider, tc.cronTickEpoch, dealId)
+					require.EqualValues(t, tc.payment, pay)
+					require.EqualValues(t, d.ProviderCollateral, slashed)
+					actor.assertDealDeleted(rt, dealId)
+
+					// running cron tick again dosen't do anything
+					actor.cronTickNoChangeBalances(rt, client, provider)
+				} else {
+					rt.ExpectAssertionFailure(tc.assertionMsg, func() {
+						rt.ExpectValidateCallerAddr(builtin.CronActorAddr)
+						rt.SetCaller(builtin.CronActorAddr, builtin.CronActorCodeID)
+						param := adt.EmptyValue{}
+						rt.Call(actor.CronTick, &param)
+						rt.Verify()
+					})
+				}
+			})
+		}
+	}
+
+	startEpoch := abi.ChainEpoch(50)
+	endEpoch := abi.ChainEpoch(300)
+
+	{
```

Comment:
Not clear why this is here. The extra indentation hurts readability a bit.

Reply:
Fixed this.